### PR TITLE
Fix authconfig instance parameters

### PR DESCRIPTION
**Pull request by averevki**
Fix to the https://github.com/Kuadrant/testsuite/pull/158. In the AuthConfig instance creation, there was still parsed string instead of the route object or list with hostnames.

### CODE REVIEW

File: testsuite/openshift/objects/auth_config/__init__.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def create_instance(cls, openshift: OpenShiftClient, name, route: Route,
                 "namespace": openshift.project
             },
             "spec": {
-                "hosts": hostnames or route.hostnames
+                "hosts": hostnames or (route.hostnames if route else None) or []
```
Comment (pehala) [inline testsuite/openshift/objects/auth_config/__init__.py:53]: why this change?

Comment (averevki) [reply testsuite/openshift/objects/auth_config/__init__.py:53]: there will be an error if hostnames and route are None. Because NoneType has no attributes

Comment (pehala) [reply testsuite/openshift/objects/auth_config/__init__.py:53]: But you need either hostnames or route, otherwise it doesnt make any sense

Comment (averevki) [reply testsuite/openshift/objects/auth_config/__init__.py:53]: yes, that's just an error treatment. there is an `add_host` function below to add hosts manually if you need it

Comment (pehala) [reply testsuite/openshift/objects/auth_config/__init__.py:53]: I would add that only after we found usecase where that makes sense, I think for now the current way is alright

**Review by pehala — APPROVED** [review]

**Review by JaurbanRH — APPROVED** [review]
